platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/counters/BackpressureObjectCounter.java

```diff
@@ -72,38 +73,43 @@ public BackpressureObjectCounter(
      */
     @Override
     public void onRamp() {
-        final long currentCount = count.get();
-        if (currentCount < capacity) {
-            final boolean success = count.compareAndSet(currentCount, currentCount + 1);
-            if (success) {
-                return;
+        while (true) {
```

Comment:
This is basically a spin until the condition is met and loop indefinitely until it is?  Does this burn 1 cpu while waiting?

Reply:
nevermind, I think I misread where the scope ends.   The brain read the closing brace, but it's in a red zone, not a green zone.

Reply:
It will burn a CPU either until the condition is met (i.e. there is capacity in the queue) or the fork join pool has sufficient resources to offload the blocked thread. This is intended as a last resort as opposed to just crashing when the exception is thrown. In practice, we shouldn't see this unless there is something hogging the fork join pool resources.